test/index.ts

```diff
@@ -1,19 +1,52 @@
 import { expect } from "chai";
 import { ethers } from "hardhat";
 
-describe("Greeter", function () {
-  it("Should return the new greeting once it's changed", async function () {
-    const Greeter = await ethers.getContractFactory("Greeter");
-    const greeter = await Greeter.deploy("Hello, world!");
-    await greeter.deployed();
+describe("BarterMarket", function () {
+  before(async function () {
```

Comment:
I think this should be before each for a fresh slate each test

Reply:
Changed this to `beforeEach`